spring-cloud-generator/src/main/java/com/google/cloud/generator/spring/composer/SpringPropertiesClassComposer.java

```diff
@@ -158,7 +150,9 @@ private static List<Statement> createMemberVariables(
             "executorThreadCount", TypeNode.INT_OBJECT, false, null, null);
     statements.add(SpringPropertiesCommentComposer.createExecutorThreadCountPropertyComment());
     statements.add(executorThreadCountVarStatement);
-    if (hasRestOption) {
+    // For GRPC+REST libraries, provide configuration property to override GRPC default and use REST transport instead
+    // For GRPC-only and REST-only libraries, this property is not provided
+    if (ComposerUtils.shouldSupportRestOptionWithGrpcDefault(transport, service)) {
       ExprStatement useRestVarStatement =
           ComposerUtils.createMemberVarStatement(
               "useRest",
```

Comment:
Have you considered a `transport` property with an enum value of "REST" or "GRPC" instead? I feel that would be more natural than `useRest`.

Reply:
Hmm, I can't recall if this was discussed when the `useRest` property was initially implemented, but two small complications I can see are:

1. This would be a breaking change for any existing users of this property (though now would be the time if it's a change we'd like to eventually make)
2. Possibly needing to document or account for case sensitivity in user-provided property strings? (e.g. REST vs. Rest)

If I'm understanding correctly, the user configuration to override would change from something like
```
<service.prefix>.useRest=true
```
to
```
<service.prefix>.transport=REST
```
This property is only exposed for libraries that support grpc+rest (and default to grpc), and having a property named `useRest` as an explicit override makes sense in my head as well, so I don't feel too strongly about either naming choice. Curious to hear any other concerns behind this suggestion though!

Reply:
We're able to make this kind of change because we're still in preview. I believe that Spring Boot's relaxed binding should be case-insensitive for enums.

Reply:
I do think explicitly specify`<service.prefix>.transport=REST` might be better in the long term, but there are more things we need to consider for this change. 
For example, what would be the default value if this property is not specified? `REST` for compute and `GRPC` for the others? What's the criteria for us to decide the default value? What if a customer configured `GRPC` for compute, are we going to error out on start up? 
In addition, are we going to support this property for existing spring-cloud-gcp products as well? Some of the service does not support `REST` yet or only support `REST`(bigquery), but I think some service like `spanner` already support both.

Reply:
> Spring Boot's relaxed binding should be case-insensitive for enums

Oh that’s good to know! I can give this change a try in a separate PR.

> For example, what would be the default value if this property is not specified? REST for compute and GRPC for the others? What's the criteria for us to decide the default value? What if a customer configured GRPC for compute, are we going to error out on start up?

@blakeli0  My understanding of this suggested change is just in name/type (`useRest` boolean => `transport` enum) rather than the underlying behaviour: the current strategy is to only provide and support this property for libraries that support both GRPC+REST transport options. For GRPC and REST-only libraries where this property is not supported, any user configuration will not do anything (unless they override the transport provider and setting beans themselves), and the client library default builders will be used.

Would there be concern in this strategy itself, or is it more that having a renamed transport property only supported and applicable to some libraries could be confusing to users?

Reply:
I think @blakeli0 brings up a good point about more things to consider. For instance, we can imagine a global `spring.cloud.gcp.transport` property that could switch all integrations to that transport. I almost feel like the property should be a list of desired transports. For instance, `GRPC,REST` would be the default and mean that choose REST only if GRPC is unavailable. Users, can still specify something like 'REST', which should fail with an unsupported-transport exception for libraries that only support GRPC.

Reply:
I feel this is a relatively large topic that worth it's own discussion, can we leave it as it is for now? And put all the discussion around the naming together in a separate doc? We can always change it in preview but we should finalize it before GA.

Reply:
For sure - I can put together a doc on this topic to further explore/discuss (similar to the other conversation on strategy to enable/disable configuration classes).